Sources/Fluent/Query/QuerySupporting.swift

```diff
@@ -254,7 +254,9 @@ public protocol QuerySupporting: Database {
     /// - parameters:
     ///     - method: Aggregate method to use.
     ///     - field: Keys to aggregate. Can be zero.
-    static func queryAggregate(_ aggregate: QueryAggregate, _ fields: [QueryKey]) -> QueryKey
+    ///     - default: Value to fall back on if the aggregate function returns NULL.
+    static func queryAggregate<C>(_ aggregate: QueryAggregate, _ fields: [QueryKey], default: C?) -> QueryKey
```

Comment:
This is, very unfortunately, a major breaking change. `QuerySupporting` is a public protocol, so changing its definition will cause projects that rely on this protocol to stop compiling. 

Some ideas around this:

*Protocol finagling*
Add this method as a _new_ protocol requirement with a default implementation that tries to fail gracefully. `SQL+QuerySupporting.swift` will still implement the updated version so everything _should_ work.

The downside here is it makes the code messy and could also not work since `SQL+QuerySupporting.swift` is also just providing extensions. Swift should try to take the more specific extension, but one can never be sure.

*Break the API*
All of Vapor's Fluent implementations use the `SQL+QuerySupporting.swift` methods, so we would definitely not break those with this change (obviously as they are being tested as a part of CI). But, it's possible other people have created Fluent driver implementations that could be broken by this change. If we were to merge it as is, we need to give them fair warning.

Reply:
Either way, we should at least offer an extension with the signature:
```swift
static func queryAggregate(_ aggregate: QueryAggregate, _ fields: [QueryKey]) -> QueryKey
```
To minimize the amount of breakage.

Reply:
@calebkleveter If I understand what you mean, any conformers to the protocol would still be required to add conformance for this new method. Even if we add an extension where it has a default value, that just helps to not break _callers_. _Conformers_ must provide methods for everything required by the protocol.

Reply:
Updated with a slightly messy (but stable) way to avoid introducing a breaking change.